### DISCUSSION TITLE
How to run Deta Base locally *without* using project key?

### DISCUSSION THREAD

**Discussion by s4m-mo** [deta/cloud-docs#451] [Help] [answered]
I love the openness of the platform and the simplicity of its practical and powerful APIs, but I am struggling to test a Deta Base locally.

For example, I don't input a private project key into `Deta()` and I don't want to commit one in my code (for obvious reasons). Also, I don't want to change the real database hosted in the cloud. How can I run a version on my local machine in development without changing the cloud one and without commiting a project key to my repository?

**Comment by jnsougata** [comment, marked as answer]
Currently local deta instance for testing doesn't exist. But if you are concerned about the project key being exposed just put that into env vars and get it form env vars inside the code
